### PR TITLE
Image list 2.0 & doc improvements

### LINKED ISSUE #1256: 2.0 - Update ImageList
This is a tracking issue for updating ImageList

**Scope:** ImageList

### PR DESCRIPTION
**Pull request by cromefire**
Fixes #1256

(plus some documentation improvements)

### CODE REVIEW

File: packages/docs/src/components/app/hljs-darcula.scss [hunk 1]
```diff
@@ -0,0 +1,79 @@
+/*
```
Comment (prateekbh) [inline packages/docs/src/components/app/hljs-darcula.scss]: why is this needed?

**Comment by cromefire** [comment]
> ^ shows that code was highlighted better earlier but now its all black. Expected?

Sorry a bit malfunctional css

File: packages/docs/src/components/app/hljs-darcula.scss (hunk 1 above)
Comment (cromefire) [reply packages/docs/src/components/app/hljs-darcula.scss]: For the dark code theme, white code in the dark theme looks ugly

**Comment by cromefire** [comment]
@prateekbh This is now ready, dark docs improvements (e.g. the site now saves your settings), test improvements, image list upgrade and a few smaller things

File: packages/docs/src/components/app/index.js [hunk 2]
```diff
@@ -196,9 +199,19 @@ const menuItems = [
...
   constructor() {
     super();
+
+    // Dark mode
+    const darkMode =
+      typeof localStorage !== 'undefined'
+        ? localStorage.getItem('dark-mode') === 'dark'
+        : false;
+    if (darkMode) {
+      document.body.classList.add('mdc-theme--dark');
```
Comment (prateekbh) [inline packages/docs/src/components/app/index.js]: NIT: may be we can just click on the switch instead?

Comment (cromefire) [reply packages/docs/src/components/app/index.js]: Yes the switch is still there, but each time you reload the page you have to put it into dark mode again. Now it's the same as before, but it remembers the users choice

**Comment by cromefire** [comment]
> Just a request I can really use smaller PR reviews though neutral_face

Yes I try but when you fix one thing, you get the next bug/issue and so on and shortly you have a bunch of fixes

This time I just improved the image list page and then fixed all the bugs caused by it

File: packages/docs/src/components/app/index.js (hunk 2 above)
Comment (cromefire) [reply packages/docs/src/components/app/index.js]: I personally find it really annoying when page becomes white again suddenly and as it an easy fix I implemented it. What do you think?

**Comment by cromefire** [comment]
Right now there only a few components left and then some structural thing that we have to decide if we want to tackle them in `2.0`

**Comment by cromefire** [comment]
And I guess all the greenkeeper branches could be delete, because there are like 50+ useless branches

File: packages/docs/src/components/app/index.js (hunk 2 above)
Comment (prateekbh) [reply packages/docs/src/components/app/index.js]: oh yeah what i mean is should we just `$(switch).click()` instead of doing this with classes?

Comment (cromefire) [reply packages/docs/src/components/app/index.js]: The only thing we would save is this line:
```typescript
document.body.classList.add('mdc-theme--dark');
```

And that approach feels kind of _cleaner_ to me, I don't like UI elements that do things on their own
